### PR TITLE
Modify project extension validate to support multiaccount extension

### PR DESCRIPTION
**Pull request by bdjilka**

### CODE REVIEW

File: connect/cli/plugins/project/extension/validations.py [hunk 1]
```diff
@@ -234,64 +269,173 @@ def validate_schedulables(config, project_dir, context):
...
+            ValidationItem(
+                'ERROR',
+                'The Web app extension class must contain at least one router '
+                'implementation function wrapped in *@router.your_method("/your_path")*.',
+                extension_class_file,
+            ),
+        )
+        return ValidationResult(messages, True)
+
+    if 'ui' not in context['descriptor']:
```
Comment (gab832) [inline connect/cli/plugins/project/extension/validations.py:376]: The ui field should not be mandatory. An extension could have a class inside webapp.py and do not have static files. What I suggest is to validate the static files ONLY if ui exists on the descriptor (what you do on lines 390-426).

File: connect/cli/plugins/project/extension/validations.py [hunk 2]
```diff
@@ -234,64 +269,173 @@ def validate_schedulables(config, project_dir, context):
...
+                f'static files: {", ".join(missed_files)}.',
+                extension_class_file,
+            ),
+        )
+        return ValidationResult(messages, True)
+
+    return ValidationResult(messages, False, context)
+
+
+def validate_anvil_extension(config, project_dir, context):
```
Comment (gab832) [inline connect/cli/plugins/project/extension/validations.py:431]: I think that here you could validate that it is used the @anvil_key_variable and as you said in the comment validate that all the variables are specified properly in the decorator.

File: connect/cli/plugins/project/extension/validations.py [hunk 3]
```diff
@@ -234,64 +269,173 @@ def validate_schedulables(config, project_dir, context):
...
-                ValidationItem(
-                    'ERROR',
-                    f'Duplicate variable name: the variable with name *{variable["name"]}* '
-                    'has already been declared.',
-                    **get_code_context(extension_class, '@variables'),
-                ),
-            )
 
-        names.append(variable["name"])
+def validate_webapp_extension(config, project_dir, context):  # noqa: CCR001
```
Comment (gab832) [inline connect/cli/plugins/project/extension/validations.py:336]: Remember that you should validate the static files defined in the extension.json exists in the static_root folder. If we define the "/static/index.html" in the extension.json we should validate that inside static_root we have the file index.html.

**Review by gab832 — CHANGES_REQUESTED** [review]
I left some comments that could improve the validation, I hope they'll be useful.

File: connect/cli/plugins/project/extension/validations.py (hunk 1 above)
Comment (marcserrat) [reply connect/cli/plugins/project/extension/validations.py:376]: that's correct, an extension can provide api and not UI but is same type at the end

File: connect/cli/plugins/project/extension/validations.py (hunk 2 above)
Comment (marcserrat) [reply connect/cli/plugins/project/extension/validations.py:431]: in decorator you point from where to extract it, not the value itself (which is optional)

**Pull request merged**